The core info cache file stores absolute paths, which prevents portability on Windows

## Description

Hey!

So I've seen 2 or 3 persons already having issues while moving/renaming their RetroArch folder on Windows. They get an error saying it couldn't load the core while mentioning the previous path they used:
```
[ERROR] Failed to open libretro core: "C:\previous\path\for\cores\core_libretro.dll"
[ERROR] Error(s): The specified module could not be found.
```
and it was REALLY confusing because `Settings > Directory > Core` was pointing to the proper location.

Turns out the cause was because `core_info.cache` stores the absolute path for the cores. So these persons had to either turn OFF the feature or delete the file to re-generate a new one.

### Expected behavior

If a core fails to load because it wasn't found, the core info cache file should be "refreshed".

### Actual behavior

Content will fail to load until core info cache is turned OFF or deleted.

### Steps to reproduce the bug

1. Make sure `Settings > Core > Cache Core Info Files` is ON.
2. Move or simply rename your RetroArch folder.
3. Try to load a game.

### Environment information

- OS: Windows, I don't think that would be an issue on other OSes?

--- a/core_info.c
+++ b/core_info.c
@@ -159,9 +159,7 @@ static bool CCJSONObjectMemberHandler(void *context, const char *pValue, size_t 
             }
             break;
          case 'p':
-            if (string_is_equal(pValue,      "path"))
-               pCtx->current_string_val      = &pCtx->core_info->path;
-            else if (string_is_equal(pValue, "permissions"))
+            if (string_is_equal(pValue,      "permissions"))
             {
                pCtx->current_string_val      = &pCtx->core_info->permissions;
                pCtx->current_string_list_val = &pCtx->core_info->permissions_list;
@@ -813,14 +811,6 @@ static bool core_info_cache_write(core_info_cache_list_t *list, const char *info
 
       rjsonwriter_add_spaces(writer, 4);
       rjsonwriter_add_start_object(writer);
-      rjsonwriter_add_newline(writer);
-
-      rjsonwriter_add_spaces(writer, 6);
-      rjsonwriter_add_string(writer, "path");
-      rjsonwriter_add_colon(writer);
-      rjsonwriter_add_space(writer);
-      rjsonwriter_add_string(writer, info->path);
-      rjsonwriter_add_comma(writer);
       rjsonwriter_add_newline(writer);
 
       rjsonwriter_add_spaces(writer, 6);
@@ -1813,18 +1803,13 @@ static core_info_list_t *core_info_list_new(const char *path,
    core_info_list->list  = core_info;
    core_info_list->count = path_list->core_list->size;
 
-#if !defined(IOS)
-   /* Read core info cache, if enabled
-    * > This functionality is hard disabled on iOS/tvOS,
-    *   where core path changes on every install
-    *   (invalidating any cached parameters) */
+   /* Read core info cache, if enabled */
    if (enable_cache)
    {
       core_info_cache_list = core_info_cache_read(info_dir);
       if (!core_info_cache_list)
          goto error;
    }
-#endif
 
    for (i = 0; i < path_list->core_list->size; i++)
    {
@@ -1851,6 +1836,12 @@ static core_info_list_t *core_info_list_new(const char *path,
          if (info_cache)
          {
             core_info_copy(info_cache, info);
+            /* Core path is 'dynamic', and cannot
+             * be cached (i.e. core directory may
+             * change between runs) */
+            if (info->path)
+               free(info->path);
+            info->path = strdup(base_path);
             /* Core lock status is 'dynamic', and
              * cannot be cached */
             info->is_locked = core_info_path_is_locked(path_list->lock_list,

--- a/menu/menu_setting.c
+++ b/menu/menu_setting.c
@@ -9726,11 +9726,6 @@ static bool setting_append_list(
 #endif
             for (i = 0; i < ARRAY_SIZE(bool_entries); i++)
             {
-#if defined(IOS)
-               if (bool_entries[i].name_enum_idx ==
-                     MENU_ENUM_LABEL_CORE_INFO_CACHE_ENABLE)
-                  continue;
-#endif
                CONFIG_BOOL(
                      list, list_info,
                      bool_entries[i].target,

